### PR TITLE
app/vmselect/promql: propagate lower bucket values when fixing a histogram

### PR DESCRIPTION
**Pull request by zekker6**
### Describe Your Changes

 In most cases histograms are exposed in sorted manner with lower buckets being first. This means that during scraping buckets with lower bounds have higher chance of being updated earlier than upper ones.

 Previously, values were propagated from upper to lower bounds, which means that in most cases that would produce results higher than expected once all buckets will become updated.
 Propagating from upper bound effectively limits highest value of histogram to the value of previous scrape. Once the data will become consistent in the subsequent evaluation this causes spikes in the result.

 Changing propagation to be from lower to higher buckets reduces value spikes in most cases due to nature of the original inconsistency.

 See: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4580

An example histogram with previous(red) and updated(blue) versions:

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: app/vmselect/promql/transform.go [hunk 1]
```diff
@@ -1061,12 +1061,12 @@ func fixBrokenBuckets(i int, xss []leTimeseries) {
 			break
 		}
 	}
-	// Substitute lower bucket values with upper values if the lower values are NaN
-	// or are bigger than the upper bucket values.
-	vNext := xss[len(xss)-1].ts.Values[i]
-	for j := len(xss) - 2; j >= 0; j-- {
+	// Substitute upper bucket values lower bucket values if the upper values are NaN
```
Comment (AndrewChubatiuk) [inline app/vmselect/promql/transform.go:1064]:
```suggestion
	// Substitute upper bucket values with lower bucket values if the upper values are NaN
```

**Comment by hagen1778** [comment]
Related tickets 
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6424 
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4768
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4580

**Review by hagen1778 — APPROVED** [review]

**Comment by valyala** [comment]
@zekker6 , @hagen1778 , please see the follow-up commit ea5c3571e99ebfa016952c98c59632a42e77077c

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.102.0 release](https://docs.victoriametrics.com/changelog/).

**Pull request merged**